### PR TITLE
switch targets to using enumerable inputs

### LINKED ISSUE #1416: Run tests using Bullseye's fancy "enumerable input" feature
From https://github.com/FakeItEasy/FakeItEasy/issues/1415

We loop over test directories manually and run the tests. Bullseye may be able to do this for us, supplying more granular output.

### PR DESCRIPTION
**Pull request by adamralph**
Fixes #1416

### CODE REVIEW

File: tools/FakeItEasy.Build/Program.cs [hunk 1]
```diff
@@ -83,36 +84,43 @@ public static void Main(string[] args)
...
+
+            Target(
+                "pack-projects",
+                DependsOn("build", "outputDirectory", "pdbgit", "get-version"),
+                forEach: ProjectsToPack,
+                action: project => Run("dotnet", $"pack {project} --configuration Release --no-build --output {OutputDirectory} /p:Version={version}"));
+
+            Target(
+                "pack-nuspecs",
+                DependsOn("build", "outputDirectory", "pdbgit", "get-version"),
```
Comment (thomaslevesque) [inline tools/FakeItEasy.Build/Program.cs]: This one doesn't actually depend on `pdbgit`. It just creates an empty package with dependencies on other packages. In fact, it doesn't even depend on `build` ;)

**Comment by blairconrad** [comment]
Dude, you're stealing my work! (#1416).

File: tools/FakeItEasy.Build/Program.cs (hunk 1 above)
Comment (adamralph) [reply tools/FakeItEasy.Build/Program.cs]: Ah, ok. I'l remove the deps on `pdbgit` and `build`.

**Comment by adamralph** [comment]
@blairconrad oh, sorry! So much for checking current issues before doing work... 😏 

I don't mind withdrawing this. Just say the word. (Or just close it.)

FWIW, I amended the commit to address @thomaslevesque's review comment.

**Comment by blairconrad** [comment]
@adamralph, no worries!

I'm content for you to continue. I had 2 substantive differences in my sandbox, though. First, I was creating the test targets from `TestSuites`:

```c#
foreach (var testSuite in TestSuites)
{
    Target(
        testSuite.Key,
        DependsOn("build", "testsDirectory"),
        forEach: testSuite.Value,
        action: testDirectory => RunTestsInDirectory(testDirectory));
}
```

We have general programming language available to us. Why not use it?
Of course this is very much a "bonus change".

The other I'll add to an actual review response.

**Comment by adamralph** [comment]
I've added a commit to generate test targets from `TestSuites`.

**Comment by blairconrad** [comment]
Thanks, @adamralph! I like it. I'll just merge before @thomaslevesque  can even see the new addition!

**Comment by adamralph** [comment]
BTW - I can't remember if we ever considered this when I was an owner, but it may be worth switching on this in the branch protection settings: 

image

I'm quite surprised it isn't switch on by default. If further changes are made after an approval then it follows that the current state of the PR has yet to be approved.

**Comment by blairconrad** [comment]
I think this is a good idea. Thanks for mentioning it, @adamralph.

**Pull request merged**